ioSPI/micrographs.py

```diff
@@ -0,0 +1,78 @@
+"""Read and write micrographs."""
+
+import os
+
+import h5py
+import mrcfile
+import numpy as np
+
+
+def populate_hdf5_with_dict(h5file, path, dic):
+    """Recursively save dictionary contents to group.
+
+    Parameters
+    ----------
+    h5file : h5py.File
+        .hdf5 file to write to.
+    path : str
+        Relative path to save dictionary contents.
+    dic : dict
+        Dictionary containing data.
+    """
+    for k, v in dic.items():
+        if isinstance(v, (np.ndarray, np.int64, np.float64, int, float, str, bytes)):
+            h5file[path + k] = v
+        elif isinstance(v, type(None)):
+            h5file[path + k] = str("None")
+        elif isinstance(v, dict):
+            populate_hdf5_with_dict(h5file, path + k + "/", v)
+        else:
+            raise ValueError("Cannot save %s type" % type(v))
+
+
+def read_micrograph_from_mrc(mrc_file):
+    """Return micrograph from an input .mrc file.
+
+    Parameters
+    ----------
+    mrc_file : str
+        File name for .mrc file to turn into micrograph
+    """
+    with mrcfile.open(mrc_file, "r", permissive=True) as mrc:
+        micrograph = mrc.data
+    if len(micrograph.shape) == 2:
+        micrograph = micrograph[np.newaxis, ...]
+    return micrograph
+
+
+def write_data_dict_to_hdf5(data_dict, h5_file):
+    """Convert arbitrary dictionary data to hdf5 file format.
+
+    Parameters
+    ----------
+    data_dict : dict
+        Dictionary of data to save.
+    h5_file : str
+        Relative path to h5 file.
+    """
+    dic = {"data": data_dict}
+    with h5py.File(h5_file, "w") as file:
+        populate_hdf5_with_dict(file, "/", dic)
+
+
+def write_mrc(output_path, projections, iterations):
```

Comment:
This is a comment a bit more granular that what I wrote it the review, but in the same vibe.

- this function should be `write_micrograph_to_mrc` for consistency in namings.
Similarly, the function `write_gemmi` in atomic_models.py should be `write_atomic_model_to_pdb`

@fredericpoitevin do you agree?

PS: These things might seem like details from a contributor perspective (yours), but I assure you it makes a lot of difference from the user's perspective :) Thanks for being patient!

Reply:
Sure I'll add those changes, no problem! I haven't looked into atomic model much as we did not do any development work on it, but I'm happy to refactor it as well given specific requests.

Reply:
In addition to the write_micrograph_to_mrc change I renamed the atomic models functions to better conform to the given standard.

Reply:
Excellent thank you!